src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.removeFromAlive();
+    }
+  }
+
+  removeFromAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
+
+  bite(damage) {
```

Comment:
[FUNCTION NAMING] - The `bite` method in the `Animal` class is misleading, as biting is specific to carnivores. This method should be removed from the `Animal` class and exist only in the `Carnivore` class.